core/jobs/batch_jobs/exploration_population_jobs.py

```diff
@@ -0,0 +1,312 @@
+        except Exception as e:
+            logging.exception(e)
+            return result.Err((exp_id, e))
+
+        return result.Ok((exp_id, exploration))
+
+    @staticmethod
+    def _update_exploration(
+        exp_model: exp_models.ExplorationModel,
+        migrated_exploration: exp_domain.Exploration,
+        exploration_changes: Sequence[exp_domain.ExplorationChange],
+        exp_rights_model: exp_models.ExplorationRightsModel
+    ) -> Sequence[base_models.BaseModel]:
+        """Generates newly updated exploration models.
+
+        Args:
+            exp_model: ExplorationModel. The exploration which
+                should be updated.
+            migrated_exploration: Exploration. The migrated exploration
+                domain object.
+            exploration_changes: sequence(ExplorationChange). The
+                exploration changes to apply.
+            exp_rights_model: ExplorationRightsModel. The
+                additional model.
+
+        Returns:
+            sequence(BaseModel). Sequence of models which should be put into
+            the datastore.
+        """
+        updated_exploration_model = (
+            exp_services.populate_exploration_model_fields(
+                exp_model, migrated_exploration))
+        commit_message = 'Added android_proto_size_in_bytes attribute.'
+        change_dicts = [change.to_dict() for change in exploration_changes]
+        with datastore_services.get_ndb_context():
+            models_to_put = updated_exploration_model.compute_models_to_commit(
+                feconf.MIGRATION_BOT_USERNAME,
+                feconf.COMMIT_TYPE_EDIT,
+                commit_message,
+                change_dicts,
+                additional_models={'rights_model': exp_rights_model}
+            ).values()
+        datastore_services.update_timestamps_multi(list(models_to_put))
+        return models_to_put
+
+    @staticmethod
+    def _generate_exploration_change(
+        exp_model: exp_models.ExplorationModel,
+        exploration: exp_domain.Exploration
+    ) -> Iterable[Tuple[str, exp_domain.ExplorationChange]]:
+        """Generates exploration change objects. Exploration change object
+        is generated if the existing model doesn't contain
+        android_proto_size_in_bytes attribute.
+
+        Args:
+            exp_model: ExplorationModel. The exploration for which to generate
+                the change objects.
+            exploration: Exploration. The exploration domain object.
+
+        Yields:
+            (str, ExplorationChange). Tuple containing exploration
+            ID and exploration change object.
+        """
+        if exp_model.android_proto_size_in_bytes is None:
+            exp_change = exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_EXPLORATION_PROPERTY,
+                'property_name': 'android_proto_size_in_bytes',
+                'new_value': exploration.android_proto_size_in_bytes
```

Comment:
Should there be an old_value here? (probably None)

Reply:
Add old value now. 
Missed earlier as it was optional part.